src/chart.zoom.js

```diff
@@ -41,32 +41,36 @@ function zoomIndexScale(scale, zoom, center) {
 
 }
 
-
 function zoomTimeScale(scale, zoom, center) {
 	var options = scale.options;
-	var minDelta, maxDelta;
-	var newDiff;
+	var range = scale.right - scale.left;
```

Comment:
I think that will change when v2.2.0 of Chart.js is released. I merged a PR that adds support for the category and time axes on the left/right